tests/dialects/test_memref.py

```diff
@@ -374,6 +375,21 @@ def test_memref_memory_space_cast():
     assert memory_space_cast.dest.type.memory_space is dest_memory_space
 
 
+def test_memref_reinterpret_cast():
+    src_type = MemRefType(i64, [4, 512])
+
+    src = TestSSAValue(src_type)
+
+    reinterpret_cast = ReinterpretCastOp.get(
+        src,
+        [TestSSAValue(IntAttr(0))],
+        [TestSSAValue(IntAttr(8)), TestSSAValue(IntAttr(256))],
+        [TestSSAValue(IntAttr(1)), TestSSAValue(IntAttr(1))],
+    )
+
+    reinterpret_cast.verify()
```

Comment:
Can you please add 
1. a normal filecheck test to` tests/f/d/m/ops.mlir`?
2. an mlir interop test to the appropriate place?
3. a filecheck test that checks for verification errors?